packages/devtools_app/lib/src/shared/preferences.dart

```diff
@@ -86,19 +86,36 @@ class InspectorPreferencesController extends DisposableController
       serviceManager.inspectorService as InspectorService;
 
   final _hoverEvalMode = ValueNotifier<bool>(false);
-  final _customPubRootDirectories = ListValueNotifier<String>([]);
+  final ListValueNotifier<String> _customPubRootDirectories =
+      ListValueNotifier<String>([]);
   final _customPubRootDirectoriesAreBusy = ValueNotifier<bool>(false);
   final _busyCounter = ValueNotifier<int>(0);
   static const _hoverEvalModeStorageId = 'inspector.hoverEvalMode';
-  static const _customPubRootDirectoriesStorageId =
+  static const _customPubRootDirectoriesStoragePrefix =
       'inspector.customPubRootDirectories';
+  String? _mainScriptDir;
+  VoidCallback? isolateListener;
+
+  Future<void> _updateMainScriptRef() async {
+    final isolateRef = serviceManager.isolateManager.mainIsolate.value!;
+    if (isolateRef.id != null) {
+      final isolate = await serviceManager.service?.getIsolate(isolateRef.id!);
+      final rootLibUri = Uri.parse(isolate?.rootLib?.uri ?? '');
+      final directorySegments = rootLibUri.pathSegments
+          .sublist(0, rootLibUri.pathSegments.length - 1);
+      final rootLibDirectory = rootLibUri.replace(
+        pathSegments: directorySegments,
+      );
+      _mainScriptDir = rootLibDirectory.path;
```

Comment:
was this code copied from somewhere in the debugger code by chance? if so, we should de-dupe and make a utility method

Reply:
ours is pretty customized, the only part that would be similar is that we are getting the rootLib.uri
In this case I think it might be more trouble to try to make a utility method